llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1377,6 +1378,21 @@ void SelectionDAGLegalize::LegalizeOp(SDNode *Node) {
   }
 }
 
+// Helper function that generates an MMO that considers the alignment of the
+// stack, and the size of the stack object
+static MachineMemOperand *getStackAlignedMMO(SDValue StackPtr,
+                                             MachineFunction &MF,
+                                             bool isObjectScalable) {
+  auto &MFI = MF.getFrameInfo();
+  int FI = cast<FrameIndexSDNode>(StackPtr)->getIndex();
+  MachinePointerInfo PtrInfo = MachinePointerInfo::getFixedStack(MF, FI);
+  auto objectSize = isObjectScalable ? ~UINT16_C(0) : MFI.getObjectSize(FI);
```

Comment:
UINT16_C should be UINT64_C. It is probably better to make sure objectSize is a uint64_t too (and use the name ObjectSize - llvm prefers capitalized variable names)

Reply:
Done!